src/mixins/suit.js

```diff
@@ -1,9 +1,19 @@
 import suit from '../util/suit';
 
-export default {
+export const createSuitMixin = ({ name }) => ({
+  props: {
+    classNames: {
+      type: Object,
+      default: undefined,
+    },
+  },
   methods: {
     suit(...args) {
-      return suit(this.widgetName, ...args);
+      const className = suit(name, ...args);
+      if (this.classNames && this.classNames[className]) {
```

Comment:
We should avoid to rely on the generated class name for the keys of the object. We should provide to the users stable keys. This way we don't break their if we choose to update our class. This is a approach really similar of what we have inside InstantSearch (see below).

https://github.com/algolia/instantsearch.js/blob/ef731cc48c7640aac71e09db799b476ef2474f3b/src/widgets/refinement-list/refinement-list.js#L271-L282

Reply:
```diff
diff --git a/src/mixins/suit.js b/src/mixins/suit.js
index af5f283..fc51857 100644
--- a/src/mixins/suit.js
+++ b/src/mixins/suit.js
@@ -8,12 +8,12 @@ export const createSuitMixin = ({ name }) => ({
     },
   },
   methods: {
-    suit(...args) {
-      const className = suit(name, ...args);
-      if (this.classNames && this.classNames[className]) {
+    suit(element, modifier) {
+      const identifier = [element, modifier].filter(Boolean).join('-');
+      if (this.classNames && this.classNames[identifier]) {
         return this.classNames[className];
       }
-      return className;
+      return suit(name, element, modifier);
     },
   },
 });
```

Does this make sense, or rather in the components themselves? I think I prefer it here, since if there's only one place it's written, there's less chance of a typo happening. 

If a typo happens in the class name, it's as much of a breaking change as in the option, and we could add custom logic here to keep it non-breaking if necessary IMO

Reply:
Use the full class then!